### PR TITLE
python310Packages.keras: 2.11.0 -> 2.12.0

### PR DESCRIPTION
**Pull request by fabaff**
Changelog: https://github.com/keras-team/keras/releases/tag/v2.12.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 229362` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages marked as broken and skipped:</summary>
  <ul>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.mask-rcnn</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.pymanopt</li>
    <li>python310Packages.pymanopt.dist</li>
    <li>python310Packages.rl-coach</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
  </ul>
</details>
<details>
  <summary>57 packages failed to build:</summary>
  <ul>
    <li>gpt2tc</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.baselines</li>
    <li>python310Packages.baselines.dist</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.edward</li>
    <li>python310Packages.edward.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.gpt-2-simple</li>
    <li>python310Packages.gpt-2-simple.dist</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.pot</li>
    <li>python310Packages.pot.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>python310Packages.tensorflow (python310Packages.tensorflow-build ,python310Packages.tensorflowWithoutCuda)</li>
    <li>python310Packages.tensorflow.dist (python310Packages.tensorflow-build.dist ,python310Packages.tensorflowWithoutCuda.dist)</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.tensorflow-probability</li>
    <li>python310Packages.tensorflow-probability.dist</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.tensorflowWithCuda.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.tflearn</li>
    <li>python310Packages.tflearn.dist</li>
    <li>python310Packages.trfl</li>
    <li>python310Packages.trfl.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>tts</li>
    <li>tts.dist</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.einops</li>
    <li>python310Packages.einops.dist</li>
    <li>python310Packages.keras</li>
    <li>python310Packages.keras.dist</li>
    <li>python310Packages.xarray-einstats</li>
    <li>python310Packages.xarray-einstats.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.einops</li>
    <li>python311Packages.einops.dist</li>
    <li>python311Packages.keras</li>
    <li>python311Packages.keras.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Update is part of #251878.
